chapters/VK_QNX_external_memory_screen_buffer/qnx_screen_buffer.adoc

```diff
@@ -0,0 +1,121 @@
+structure in the pname:pNext chain.
+Images can: be created with an external format even if the QNX Screen
+buffer has a format which has an
+<<memory-external-qnx-screen-buffer-formats,equivalent Vulkan format>>
+to enable consistent handling of images from sources that might use either
+category of format.
+The external format of a QNX Screen buffer can: be obtained by
+passing a slink:VkScreenBufferFormatPropertiesQNX structure to
+flink:vkGetScreenBufferPropertiesQNX.
+
+
+[[memory-external-qnx-screen-buffer-image-resources]]
+===== QNX Screen Buffer Image Resources
+
+QNX Screen buffers have intrinsic width, height, format, and usage
+properties, so Vulkan images bound to memory imported from a QNX Screen
+buffer must: use dedicated allocations:
+sname:VkMemoryDedicatedRequirements::pname:requiresDedicatedAllocation must:
+be ename:VK_TRUE for images created with
+slink:VkExternalMemoryImageCreateInfo::pname:handleTypes that includes
+ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_SCREEN_BUFFER_BIT_QNX.
+When creating an image that will be bound to an imported QNX Screen buffer,
+the image creation parameters must: be equivalent to the
+code:_screen_buffer properties as described by the valid usage of
+slink:VkMemoryAllocateInfo.
+
+[[memory-external-qnx-screen-buffer-formats]]
+.QNX Screen Buffer Format Equivalence
+[width="100%",options="header"]
+|====
+| QNX Screen Format                              | Vulkan Format
+| code:SCREEN_FORMAT_RGBA8888                    | ename:VK_FORMAT_B8G8R8A8_UNORM
+| code:SCREEN_FORMAT_RGBX8888 ^1^                | ename:VK_FORMAT_B8G8R8A8_UNORM
+| code:SCREEN_FORMAT_BGRA8888                    | ename:VK_FORMAT_R8G8B8A8_UNORM
+| code:SCREEN_FORMAT_BGRX8888 ^1^                | ename:VK_FORMAT_R8G8B8A8_UNORM
+| code:SCREEN_FORMAT_RGBA1010102                 | ename:VK_FORMAT_A2R10G10B10_UNORM_PACK32
+| code:SCREEN_FORMAT_RGBX1010102 ^1^             | ename:VK_FORMAT_A2R10G10B10_UNORM_PACK32
+| code:SCREEN_FORMAT_BGRA1010102                 | ename:VK_FORMAT_A2B10G10R10_UNORM_PACK32
+| code:SCREEN_FORMAT_BGRX1010102 ^1^             | ename:VK_FORMAT_A2B10G10R10_UNORM_PACK32
+| code:SCREEN_FORMAT_RGBA5551                    | ename:VK_FORMAT_A1R5G5B5_UNORM_PACK16
+| code:SCREEN_FORMAT_RGBX5551 ^1^                | ename:VK_FORMAT_A1R5G5B5_UNORM_PACK16
+| code:SCREEN_FORMAT_RGB565                      | ename:VK_FORMAT_R5G6B5_UNORM_PACK16
+| code:SCREEN_FORMAT_RGB888                      | ename:VK_FORMAT_R8G8B8_UNORM
+|====
+
+
+1::
+    Vulkan does not differentiate between
+    code:SCREEN_FORMAT_RGBA8888 and
+    code:SCREEN_FORMAT_RGBX8888: they both behave as
+    ename:VK_FORMAT_R8G8B8A8_UNORM.
+    After an external entity writes to a
+    code:SCREEN_FORMAT_RGBX8888 QNX Screen buffer, the
+    values read by Vulkan from the X/A component are undefined:.
+    To emulate the traditional behavior of the X component during sampling
+    or blending, applications should: use ename:VK_COMPONENT_SWIZZLE_ONE in
```

Comment:
this seems can be done by driver internally for the x component ?

Reply:
> Is it necessory to introduce new structure VkScreenBufferPropertiesQNX?
since it is not used by application, driver implementation should be able to query these properties from screen buffer object directly, I think we can remove it completely.
vkGetScreenBufferPropertiesQNX can be removed as well, correct ?

That is a good point. However, recall that I said that there is no explicit way to determine the actual, underlying allocationSize explicitly from Screen APIs. For this reason, and because we are following the convention of the similar Android Hardware Buffer Extension (see subsequent quote), we'd like to keep these available as a way to query information directly from a Screen native buffer -- (i.e. for VkMemoryAllocateInfo: "If the parameters define an import operation and the external handle type is VK_EXTERNAL_MEMORY_HANDLE_TYPE_ANDROID_HARDWARE_BUFFER_BIT_ANDROID, allocationSize must be the size returned by vkGetAndroidHardwareBufferPropertiesANDROID()).

Reply:
> this seems can be done by driver internally for the x component ?

This should only be done if requrested explicitly by the Vulkan user. This comment is only saying that these two formats will be treated equivalently by the Vulkan implementation, and if somebody wants to emulate the RGBX8888 behavior, they should use the Vulkan swizzle/blend methods explicitly.

Reply:
OK, sounds good, thanks for the explaination.